src/wic/api/pythonapi.py

```diff
@@ -155,61 +167,80 @@ def _(val: bool) -> bool:
     return val
 
 
-class Step(Tool):  # pylint: disable=too-few-public-methods
-    """Base class for configured CLTs."""
+class Step(BaseModel):  # pylint: disable=too-few-public-methods
+    """Base class for Step of Workflow."""
+    if PYDANTIC_V2:
+        model_config: ClassVar[ConfigDict] = ConfigDict(arbitrary_types_allowed=True)
+    else:
+        class Config:  # pylint: disable=too-few-public-methods
+            arbitrary_types_allowed = True
 
-    cwl_name: str
+    clt: CWLCommandLineTool
+    clt_path: Path
+    clt_name: str
+    cwl_version: str
     inputs: list[CLTInput]
     yaml: dict[str, Any]
     cfg_yaml: dict = Field(default_factory=_default_dict)
-    _input_names: list[str]
+    _input_names: list[str] = PrivateAttr(default_factory=list)
 
-    def __init__(self, cwl_path: StrPath, config_path: Optional[Path] = None):
+    def __init__(self, clt_path: StrPath, config_path: Optional[StrPath] = None):
         # validate using cwl.utils
-        if not isinstance(cwl_path, (Path, str)):
+        if not isinstance(clt_path, (Path, str)):
             raise TypeError("cwl_path must be a Path or str")
-        if isinstance(cwl_path, str):
-            cwl_path_ = Path(cwl_path)
-        else:
-            cwl_path_ = cwl_path
+        clt_path_ = Path(clt_path) if isinstance(clt_path, str) else clt_path
         try:
-            cwl = load_document_by_uri(cwl_path_)
+            clt = load_document_by_uri(clt_path_)
```

Comment:
This line of code may not do what you think it does. Please see the validate_cwl function here:
https://github.com/PolusAI/workflow-inference-compiler/blob/master/src/wic/plugins.py#L57-L61

It is very easy to look at tutorials and use the seemingly-straightforward code from them, but there are often nasty subtle edge cases that need to be considered (and have already been painfully discovered and solved!).

Moreover, there is an even nastier issue in this case w.r.t. skip_schemas, which you may discover if you turn on your VPN.

So let's try to explore the codebase and make ourselves aware of the existing production-ready code.